test/cmdlineTests.sh

```diff
@@ -112,8 +112,10 @@ function test_solc_behaviour()
         sed -i.bak -e 's/{[^{]*Warning: This is a pre-release compiler version[^}]*},\{0,1\}//' "$stdout_path"
         sed -i.bak -E -e 's/ Consider adding \\"pragma solidity \^[0-9.]*;\\"//g' "$stdout_path"
         sed -i.bak -e 's/"errors":\[\],\{0,1\}//' "$stdout_path"
-        # Remove explicit bytecode and references to bytecode offsets
-        sed -i.bak -E -e 's/\"object\":\"[a-f0-9]+\"/\"object\":\"bytecode removed\"/g' "$stdout_path"
+        # Remove medatata from bytecode bytecode and references to bytecode offsets
+        # 64697066735822 = hex encoding of 0x64 'i' 'p' 'f' 's' 0x58 0x22
+        # 64736f6c63     = hex encoding of 0x64 's' 'o' 'l' 'c'
+        sed -i.bak -E -e 's/(\"object\":\"[^"]*64697066735822)[0-9a-f]+(64736f6c63)[0-9a-f]+([^"]*\")/\1<IPFS HASH REMOVED>\2<COMPILER VERSION REMOVED>\3/g' "$stdout_path"
```

Comment:
Can you use some lookahead option in the regexp, e.g. find the last 2 bytes and trunacate by that much?

Or just use some other tool to do that?

Reply:
Maybe. But what would be the goal be? It would not remove the assumption that the version part continues until the end of the binary (except for those two bytes).